[TDX-5270] Snippets docs (#8672)
Kong/docs.konghq.com: app/dev-portal/portals/customization/snippets.md

---
title: Snippets
---

> Snippets enable reuse of content and MDC components across [Pages](/dev-portal/portals/customization/custom-pages). Visibility controls and Publishing status allow you to enable/disable Snippets, and/or restrict access to logged-in Developers.

Reviewer: ```suggestion
Snippets allow you to reuse content and MDC components across [Pages](/dev-portal/portals/customization/custom-pages). You can enable or disable snippets with visibility controls and publishing status. You can also restrict access to logged in developers.
```

> To get started creating Snippets, navigate to your Dev Portal and select [**Portal Editor**](/dev-portal/portals/customization/portal-editor/), then Snippets.

Reviewer: ```suggestion
To get started creating Snippets, navigate to your Dev Portal and click [**Portal editor**](/dev-portal/portals/customization/portal-editor/), then click **Snippets**.
```

> To get started creating Snippets, navigate to your Dev Portal and select [**Portal Editor**](/dev-portal/portals/customization/portal-editor/), then Snippets from the left sidebar.

Reviewer 2: > then Snippets from the left sidebar

This navigation will be changing as Travis mentioned

Author: Removed "from the left sidebar" to be a bit more ambiguous for the moment

{:.note}

> > *Snippets are built using Markdown Components (MDC). Additional documentation on Snippet syntax and usage is available on a [dedicated MDC site](https://portaldocs.konghq.com/components/snippet).*

Reviewer: ```suggestion
> Snippets are built using Markdown Components (MDC). See the [dedicated MDC site](https://portaldocs.konghq.com/components/snippet) for more information about Snippet syntax and usage.
```

> ### Create a new Snippet

Reviewer: ```suggestion
## Create a new Snippet
```

> * Click `+` at the top

Reviewer 2: suggestion: similar to what Travis mentioned, can we just reference "Click to create a new Snippet" instead of specific instructions in case the UX changes over time?

* Give the Snippet a "name". This will be used to refer to your Snippet from Pages.

> * The Snippet will be created with the `title` in front matter set to the specified name. Snippet front matter is not used in generating Pages, but can be useful to keep track of what you're working on alongside other Editors.

Reviewer 2: ```suggestion
* The Snippet will be created with the `title` in front matter set to the specified name. Snippet front matter can be useful to keep track of what you're working on alongside other Editors as well as providing additional data to the Snippet.
```

Author: Can you clarify how Snippet front matter is used in generating the Page? Should I refine to `title`/`description` isn't used?

Reviewer 2: To be honest I wouldn't specify anything "isn't used" for now since we may purpose these properties for something in the near future.

Snippet front matter _is used_ when properties are utilized within the Snippet, so I think the current verbiage (without my changes) is confusing.

> * Click to create a new Snippet at the top of the left sidebar.
> * Give the Snippet a "name". This will be used to refer to your Snippet from Pages, and must be a unique lowercase, `kebab-case` string.
> * The Snippet will be created with the `title` in front matter set to the specified name. Snippet front matter is not used in generating Pages, but can be useful to keep track of what you're working on alongside other Editors.

Reviewer: ```suggestion
1. In the Dev Portal editor sidebar, click to create a new Snippet.
1. Give the Snippet a name. This will be used to refer to your Snippet from Pages, and must be a unique lowercase string, for example: `my-snippet`.
1. The Snippet will be created with the `title` in the front matter set to the specified name. The Snippet front matter isn't used in generating Pages, but can be useful to keep track of what you're working on alongside other editors.
```

> * Edit the content of your Snippet in the middle panel using any Markdown or MDC, and you'll see a Live Preview in the right panel.

Reviewer: ```suggestion
1. Edit the content of your Snippet in the pane using any Markdown or MDC, and you'll see a live preview in the right pane.
```

{:.important}
> *Preview may not always be able to display parameterized values. When the Page is being rendered in your Portal, parameters will be resolved. Depending on the type of syntax used, the Preview may not reflect parameterized values in Page or Snippet views.*

{:.note}

> > *Snippets are `Published` by default for ease of use. Unilt the Snippet has been referred to in a Page, it will not be displayed in the rendered portal.*

Reviewer 2: typo
```suggestion
> *Snippets are `Published` by default for ease of use. Until the Snippet has been referred to in a Page, it will not be displayed in the rendered portal.*
```

{:.note}
> *Snippets will be created with the Default Page Visibility as set in your [Settings](/dev-portal/portals/settings/general)*

{:.important}

> > Snippets are limited to 1 000 000 characters.

Reviewer 2: Where is this limit enforced?

Author: Dochub, Benjamin clarified this for Pages, I assume it would be the same for Snippets

Reviewer 2: Ah ok yep

> {:.important}
> > *Preview may not always be able to display parameterized values. When the Page is being rendered in your Portal, parameters will be resolved. Depending on the type of syntax used, the Preview may not reflect parameterized values in Page or Snippet views.*
> 
> {:.note}
> > *Snippets are `Published` by default for ease of use. Until the Snippet has been referred to in a Page, it will not be displayed in the rendered portal.*
> 
> {:.note}
> > *Snippets will be created with the Default Page Visibility as set in your [Settings](/dev-portal/portals/settings/general)*
> 
> {:.important}
> > Snippets are limited to 1,000,000 characters.

Reviewer: ```suggestion
{:.important}
> **Notes:**
> * Preview may not always be able to display parameterized values. When the Page is rendered in your Portal, parameters will be resolved. Depending on the type of syntax used, the preview may not reflect parameterized values in Page or Snippet views.
> * Snippets are published by default. The Snippet won't be displayed in your Portal until it's reused in a Page.
> * Snippets are created with the default page visibility that is set in your [settings](/dev-portal/portals/settings/general).
> * Snippets are limited to 1,000,000 characters.
```

> ### Reference a Snippet within a Page

Reviewer: ```suggestion
## Reference a Snippet in a Page
```

> In it's simplest usage, use the `Snippet` component within a page, and specify the `name` of the Snippet. These properties are autocompleted from the list of your previously created Snippets.

Reviewer 2: typo
```suggestion
In its simplest usage, use the `Snippet` component within a Page, and specify the `name` of the Snippet. These properties are auto-completed from the list of your previously created Snippets.
```

> In its simplest usage, use the `Snippet` component within a Page, and specify the `name` of the Snippet. These properties are auto-completed from the list of your previously created Snippets.

Reviewer: ```suggestion
You can reuse the Snippet component within a Page by specifying the name of the Snippet. These properties are auto-completed from the list of your previously created Snippets.

For example:
```

```mdc
::snippet
---
name: "get-api-keys"
---
::
```

For more advanced usage, including passing data into Snippets, see our [dedicated MDC site](https://portaldocs.konghq.com/components/snippet).

### Modify a Snippet

In the middle panel, you can make changes to your MDC content, and instantly see the live Preview.

Once you have completed your changes, be sure to click **Save**.

{:.note}

> > *To rename a snippet's title, click the 'three dots' menu, and click Rename*

Reviewer 2: Rename, not change title
```suggestion
> *To rename a Snippet, click the 'three dots' menu, and click Rename*
```

> ### Modify a Snippet
> 
> In the middle panel, you can make changes to your MDC content, and instantly see the live Preview.
> 
> Once you have completed your changes, be sure to click **Save**.
> 
> {:.note}
> > *To rename a Snippet, click the 'three dots' menu, and click Rename*

Reviewer: ```suggestion
```
I'd remove this since this UX flow is common, so it doesn't need to be explained.

### Publishing Snippets

> Newly created pages are in `Published` status by default for ease of use. With the Snippet selected on the left sidebar, click the "three dots" menu in the top right corner and select `Unpublish`. This can be useful for providing messaging across Pages that is only displayed for a period of time, e.g. system outages or special events.

Reviewer 2: ```suggestion
Newly created Pages are in `Published` status by default for ease of use. With the Snippet selected on the left sidebar, click the "three dots" menu in the top right corner and select `Unpublish`. This can be useful for providing messaging across Pages that is only displayed for a period of time, e.g. system outages or special events.
```

> ### Publishing Snippets
> 
> Newly created Pages are in `Published` status by default for ease of use. With the Snippet selected on the left sidebar, click the "three dots" menu in the top right corner and select `Unpublish`. This can be useful for providing messaging across Pages that is only displayed for a period of time, e.g. system outages or special events.

Reviewer: ```suggestion
## Unpublishing Snippets

Newly created Pages are published by default. If you want to unpublish a Snippet, select the Snippet in the sidebar and click the menu in the top right corner and select **Unpublish**. 

This can be useful for providing messaging across Pages that is only displayed for a period of time, for example, system outages or special events.
```

> ### Change Page visibility
> 
> In the top right corner, click the menu with three dots. You can toggle from `Private` to `Public`, and vice versa.

Reviewer: ```suggestion
## Change Page visibility
If you want to change the visibility of a Snippet, select the Snippet in the sidebar and click the menu in the top right corner and toggle **Private** and **Public** as needed.
```
